modules/tiles/src/tileset/tileset-3d.ts

```diff
@@ -366,13 +365,19 @@ export default class Tileset3D {
    * Update visible tiles relying on a list of viewports
    * @param viewports - list of viewports
    */
+  // eslint-disable-next-line max-statements, complexity
   update(viewports: any[]): void {
     if ('loadTiles' in this.options && !this.options.loadTiles) {
       return;
     }
     if (this.traverseCounter > 0) {
       return;
     }
+    if (!viewports && this.lastUpdatedVieports) {
```

Comment:
This looks odd: Per the typescript types, `viewports` should never be able to be `falsy`. At worst is could be an empty array...

Is this called from some code that is not type checked?

Come to think of it, there are some i3s related excludes at the bottom of `tsconfig.json` - you may want to start removing those and fixing the resulting errors.

Reply:
It is called from debug application. See the change in `app-debug.js`. It is done to raise re-render even if `viewportsState` doesn't change.